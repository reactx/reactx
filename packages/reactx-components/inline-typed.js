/**
 * Copyright (c) ReactX and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

// This file must have the Flow annotation.
//
// This is the Flow-typed entry point for the reconciler. It should not be
// imported directly in code. Instead, our Flow configuration uses this entry
// point for the currently checked renderer (the one you passed to `yarn flow`).

export type ButtonProps = {
  id?: string,
  title?: string,
  type?: string,
  btncolor?: string,
  disabled: boolean,
  badgeProps: {
    badge: string,
    className: string,
    badgecolor: string,
  },

  buttonProps: any,
  className?: string,
  forwardedRef: {current: any},
  renderButton?: (props: any) => void,
  renderBadge?: (props: any) => void,
  onClick: (e: MouseEventHandler<T>) => void,
};
export * from './src/ReactComponents';
