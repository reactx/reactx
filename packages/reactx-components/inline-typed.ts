/**
 * Copyright (c) ReactX and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 
 */

export type BaseColor =
  | 'none'
  | 'normal'
  | 'primary'
  | 'success'
  | 'danger'
  | 'warning'
  | 'info'
  | 'dark'
  | 'light';

export type LoadingSizeType = 'tiny' | 'small' | 'medium' | 'large' | 'extra';
