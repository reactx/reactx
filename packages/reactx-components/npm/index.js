'use strict';

if (process.env.NODE_ENV === 'production') {
  module.exports = require('./reactx-components.min.js');
} else {
  module.exports = require('./reactx-components.js');
}
