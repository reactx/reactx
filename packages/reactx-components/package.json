{
  "name": "@reactx/reactx-components",
  "description": "The standard React Components",
  "version": "1.0.20",
  "author": "Behzad Abbasi <overafaik.com>",
  "bugs": {
    "url": "https://github.com/reactx/reactx/issues"
  },
  "homepage": "https://github.com/reactx/reactx/tree/master/packages/reactx-components#readme",
  "keywords": [
    "reactx",
    "components"
  ],
  "license": "MIT",
  "files": [
    "LICENSE",
    "README.md",
    "index.js",
    "index.d.ts",
    ".npmrc",
    "reactx-components.js",
    "reactx-components.css",
    "reactx-components.min.js",
    "reactx-components.min.css",
    "src/"
  ],
  "main": "./index.ts",
  "typings": "./index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/reactx/reactx.git",
    "directory": "packages/reactx-components"
  },
  "dependencies": {
    "@ag-grid-community/core": "^26.0.0",
    "@ag-grid-community/all-modules": "^26.0.0",
    "@ag-grid-enterprise/core": "^26.0.0",
    "@ag-grid-enterprise/all-modules": "^26.0.0",
    "@ag-grid-community/react": "^26.0.0",
    "react-datepicker": "^4.2.1",
    "date-fns": "^2.24.0",
    "@popperjs/core": "^2.10.1",
    "react-transition-group": "^4.4.2",
    "classnames": "^2.3.1"
  }
}
