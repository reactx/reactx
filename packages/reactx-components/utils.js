export const cleanProps = (props) => {
  Object.keys(props).forEach(function (key) {
    if (typeof props[key] === 'undefined' || props[key] === null) {
      delete props[key];
    }
  });
};
