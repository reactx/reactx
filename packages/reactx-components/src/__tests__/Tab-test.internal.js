/**
 * Copyright (c) ReactX and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 *
 */

'use strict';
import React from 'react';
import ReactDOM from 'react-dom';
import {Tabs, Tab, TabPanel} from 'reactx-components';

describe('while running in a browser environment', () => {
  let container, wrapper, onSetOpen;

  beforeEach(() => {
    jest.resetModules();
    onSetOpen = jest.fn();
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    ReactDOM.render(null, container);
    document.body.removeChild(container);
    container = null;
  });

  it('should render correctly', () => {
    wrapper = (
      <div>
        <Tabs value={0}>
          <Tab label={'First'} id={0}></Tab>
          <Tab label={'Second'} id={1}></Tab>
          <Tab label={'Third'} id={2}></Tab>
          <TabPanel index={0}>First Panel</TabPanel>
          <TabPanel index={1}>Second Panel</TabPanel>
          <TabPanel index={2}>Third Panel</TabPanel>
        </Tabs>
      </div>
    );

    expect(ReactDOM.render(wrapper, container)).toMatchSnapshot();
  });
});
