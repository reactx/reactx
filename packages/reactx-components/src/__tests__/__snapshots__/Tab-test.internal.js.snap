// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while running in a browser environment should render correctly 1`] = `
<div>
  <ul
    value="0"
  >
    <li
      class=" selected"
      id="0"
      label="First"
    >
      First
    </li>
    <li
      class=" "
      id="1"
      label="Second"
    >
      Second
    </li>
    <li
      class=" "
      id="2"
      label="Third"
    >
      Third
    </li>
  </ul>
  <div
    class=" undefined"
    index="0"
    label="Label"
  >
    First Panel
  </div>
  <div
    class=" undefined"
    index="1"
    label="Label"
  />
  <div
    class=" undefined"
    index="2"
    label="Label"
  />
</div>
`;
